plugins/action/common/prepare_plugins/prep_108_route_control.py

```diff
@@ -0,0 +1,298 @@
+            80: "www",
+            101: "hostname",
+            109: "pop2",
+            110: "pop3",
+            111: "sunrpc",
+            113: "ident",
+            119: "nntp",
+            179: "bgp",
+            194: "irc",
+            496: "pim-auto-rp",
+            512: "exec",
+            513: "login",
+            514: "cmd",
+            515: "lpd",
+            517: "talk",
+            540: "uucp",
+            543: "klogin",
+            544: "kshell",
+            3949: "drip",
+        }
+
+        udp = {
+            7: "echo",
+            9: "discard",
+            37: "time",
+            42: "nameserver",
+            49: "tacacs",
+            53: "domain",
+            67: "bootps",
+            68: "bootpc",
+            69: "tftp",
+            111: "sunrpc",
+            123: "ntp",
+            137: "netbios-ns",
+            138: "netbios-dgm",
+            139: "netbios-ss",
+            161: "snmp",
+            162: "snmptrap",
+            177: "xdmcp",
+            195: "dnsix",
+            434: "mobile-ip",
+            496: "pim-auto-rp",
+            500: "isakmp",
+            512: "biff",
+            513: "who",
+            514: "syslog",
+            517: "talk",
+            520: "rip",
+            4500: "non500-isakmp",
+        }
+
+        if protocol == 'tcp':
+            if port_number in tcp:
+                return tcp[port_number]
+
+        elif protocol == 'udp':
+            if port_number in udp:
+                return udp[port_number]
+
+        return port_number
+
+    def rewrite_match_interface(self, interfaces):
+        """
+        Rewrite interface with proper case and name
+        """
+        new_interfaces = []
+        for interface in interfaces:
+            intf_ethernet = re.match(r'(?i)^(?:e|eth(?:ernet)?)(\d(?:\/\d+){1,2}$)', interface)
+            intf_ethernetdot1q = re.match(r'(?i)^(?:e|eth(?:ernet)?)(\d(?:\/\d+){1,2}.(\d+)$)', interface)
+            intf_loopback = re.match(r'(?i)^(?:lo|lo(?:opback)?)(\d+)$', interface)
+            intf_portchannel = re.match(r'(?i)^(?:po|po(?:rt-channel)?)(\d+)$', interface)
+            intf_portchanneldot1q = re.match(r'(?i)^(?:po|po(?:ort-channel)?)(\d{1,4}.\d+)$', interface)
+            intf_null = re.match(r'(?i)^(?:n|null:?)([0])$', interface)
+            if intf_ethernet:
+                new_interfaces.append("Ethernet" + intf_ethernet[1])
+            elif intf_ethernetdot1q:
+                new_interfaces.append("Ethernet" + intf_ethernetdot1q[1])
+            elif intf_loopback:
+                new_interfaces.append("loopback" + intf_loopback[1])
+            elif intf_portchannel:
+                new_interfaces.append("port-channel" + intf_portchannel[1])
+            elif intf_portchanneldot1q:
+                new_interfaces.append("port-channel" + intf_portchanneldot1q[1])
+            elif intf_null:
+                new_interfaces.append("Null" + intf_null[1])
+        return new_interfaces
```

Comment:
Why are you having to do this extra functional code vs just ignoring case where you call this helper function?

Reply:
Schema allows interfaces with different cases and short or full name. However in the CLI, `Ethernet` and `Null` interfaces have a Capital letter, which is not the case for `port-channel` and `loopback`.

This function reformat interface names with the full name and the good case.

Ex:
```ios
  route-map rm-interfaces permit 10
      description Match interfaces
      match interface Ethernet1/10 loopback100 Null0 port-channel100
```

```yaml

---
vxlan:
  overlay_extensions:
    route_control:
      route_maps:
        - name: rm-interfaces
          entries:
            - seq_number: 10
              operation: permit
              description: Match interfaces
              match:
                - interfaces:
                    - Ethernet1/10
                    - loopback100
                    - null0
                    - po100
      groups:
        - name: leaf1
          route_maps:
            - name: rm-interfaces
      switches:
        - name: netascode-leaf1
          groups: leaf1
```

Reply:
Using Jinja filters wouldn't it be easier to match on ```eth``` for example and use the title filter vs having to maintain a mapper? Same applies to Null. Also, same could be applied with matching ```po``` or ```port``` for port-channel and use the lower filter. Thoughts???

cc @mikewiebe

Reply:
Currently I updated on the python script and join all interfaces in Jinja:

Ex:
```
  match interface {{ match["interface"] | join(" ") }}
```

Should I move all in Jinja ?